src/server/search/search_family.cc

```diff
@@ -1131,18 +1131,59 @@ void SearchFamily::FtAlter(CmdArgList args, const CommandContext& cmd_cntx) {
 
 void SearchFamily::FtDropIndex(CmdArgList args, const CommandContext& cmd_cntx) {
   string_view idx_name = ArgS(args, 0);
-  // TODO: Handle optional DD param
 
+  // Parse optional DD (Delete Documents) parameter
+  bool delete_docs = false;
+  if (args.size() > 1) {
+    string_view option = ArgS(args, 1);
+    // Only check for DD option, ignore other arguments for compatibility
+    if (absl::EqualsIgnoreCase(option, "DD")) {
+      delete_docs = true;
+    }
+  }
+
+  // Collect document keys (if DD is set), drop index, and delete documents in single transaction
+  vector<vector<string>> shard_doc_keys(delete_docs ? shard_set->size() : 0);
```

Comment:
You don't need this inside the command, you can store the keys inside the transaction callback locally